Network/AMQP/Internal.hs

```diff
@@ -518,7 +531,7 @@ channelReceiver chan = do
             )
     handleAsync (SimpleMethod (Channel_close _ (ShortString errorMsg) _ _)) = do
         closeChannel' chan errorMsg
-        killThread =<< myThreadId
+        myThreadId >>= flip CE.throwTo (ConnectionClosedException . T.unpack $ errorMsg)
```

Comment:
Using throwIO would be slightly shorter :) But it probably won't matter in the end.